neqo-bin/src/server/http09.rs

```diff
@@ -29,29 +27,44 @@ pub struct HttpServer {
     server: Server,
     write_state: HashMap<StreamId, HttpStreamState>,
     read_state: HashMap<StreamId, Vec<u8>>,
+    is_qns_test: bool,
 }
 
 impl HttpServer {
     pub fn new(
-        now: Instant,
-        certs: &[impl AsRef<str>],
-        protocols: &[impl AsRef<str>],
+        args: &Args,
         anti_replay: AntiReplay,
         cid_manager: Rc<RefCell<dyn ConnectionIdGenerator>>,
-        conn_params: ConnectionParameters,
     ) -> Result<Self, Error> {
+        let mut server = Server::new(
+            args.now(),
+            &[args.key.clone()],
+            &[args.shared.alpn.clone()],
+            anti_replay,
+            Box::new(AllowZeroRtt {}),
+            cid_manager,
+            args.shared.quic_parameters.get(&args.shared.alpn),
+        )?;
+
+        server.set_ciphers(&args.get_ciphers());
+        server.set_qlog_dir(args.shared.qlog_dir.clone());
+        if args.retry {
+            server.set_validation(ValidateAddress::Always);
+        }
+        if args.ech {
+            let (sk, pk) = generate_ech_keys().expect("generate ECH keys");
+            server
+                .enable_ech(random::<1>()[0], "public.example", &sk, &pk)
+                .expect("enable ECH");
+            let cfg = server.ech_config();
+            qinfo!("ECHConfigList: {}", hex(cfg));
+        }
```

Comment:
Moved from `HttpServer` `trait` `fn` implementations (see deletes below) into `http09::HttpServer` `struct` `new()` in order to keep the `HttpServer` `trait` lean. By keeping it lean the `HttpServer` trait only requires what is needed in both Neqo's server implementation and mozilla-centrals server implementation.